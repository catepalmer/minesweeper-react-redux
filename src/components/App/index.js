import React from 'react'
import styled from 'styled-components'
import { contains, times } from 'ramda'

import { Board, Square } from '..'
import { checkIfPlayed, getMinesTouching, setMines } from '../../utilities'

const mines = setMines()

function makeSquares (mines, moves) {
  return times(square => {
    const isPlayed = checkIfPlayed(square, moves)

    return (isPlayed === false)
      ? <Square
        key={square}
        handleClick={() => console.log(`Square ${square}`)}
        index={square}
        />
        : <Square
        key={square}
        index={square}
        isMine={contains(square, mines) ? true : false}
        minesTouching={contains(square, mines) ? null : getMinesTouching(square, mines)}
        />
  }, 81)
}

const StyledApp = styled.div`
  display: grid;
  font-family: 'Verdana', sans-serif;
  grid-template-areas: 'board';
  height: 100vh;
  margin: 0;
  padding: 0;
  width: 100vw;
`
StyledApp.displayName = 'StyledApp'

export default function App ({ moves = [] }) {
  return (
    <StyledApp>
      <Board>
        {makeSquares(mines, moves)}
      </Board>
    </StyledApp>
  )
}