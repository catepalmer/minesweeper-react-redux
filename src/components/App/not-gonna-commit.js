const mines = []

function listMines () {
    for (let i = 0; mines.length < 10; i++) {
        let mineIndex = Math.floor(Math.random() * 81)
        if (mines.find(mine => mine === mineIndex)) return
        mines.push(mineIndex)
    }
    return mines
}




function makeMines (mines = []) {
  if (mines.length === 10) return mines
  const idx = Math.floor(Math.random() * 81)
  if (R.not(R.contains(idx, mines))) mines.push(idx)
  return makeMines(mines)
}



const makeMines = (mines = [], idx = Math.floor(Math.random() * 81)) => {
  if (mines.length === 10) return mines
  if (mines.includes(idx) === false) mines.push(idx)
  return makeMines(mines)
}


const makeMines = (mines = [], idx = Math.floor(Math.random() * 81)) => {
  if (R.length(mines) === 10) return mines
  if (R.not(R.contains(idx, mines))) mines.push(idx)
  return makeMines(mines)
}



// while there are fewer than 10 mines in the mines array,
  // generate a random number between 0 and 80 and, if this
  // number isn't already in the mines array, add it to the
  // mines array.  



import { head, inc, negate, reduceRight } from 'ramda'

const compose = (...funcs) => {
  return (...args) => head(
    reduceRight((acc, f) => [f(...acc)], args, funcs)
  )
}