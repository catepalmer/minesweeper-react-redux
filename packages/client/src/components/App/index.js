import React from 'react'
import styled from 'styled-components'
import { times } from 'ramda'

import { Board } from '..'
import { NewGameButton, Square } from '../../containers'

import { setMines } from '../../utilities'

const StyledApp = styled.div`
  display: grid;
  font-family: 'Verdana', sans-serif;
  grid-template-areas: 'board';
  height: 100vh;
  margin: 0;
  padding: 0;
  width: 100vw;
`
StyledApp.displayName = 'StyledApp'

export default function App () {
  const mines = setMines()
  console.log(`mines from App component: ${mines}`)
  
  return (
    <StyledApp mines={mines}>
      <Board>
        {times(square => <Square key={square} index={square} mine={mines} />, 81)}
      </Board>
      <NewGameButton mines={mines} />
    </StyledApp>
  )
}