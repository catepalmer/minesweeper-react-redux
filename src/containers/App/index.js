import { connect } from 'react-redux'

import App from '../../components/App'
import { minesSet } from '../../state'
import { setMines } from '../../utilities'

const mines = setMines()

function mapDispatchToProps(dispatch) {

  return {
    mines: () => dispatch(minesSet(mines))
  }
}

export default connect(null, mapDispatchToProps)(App)