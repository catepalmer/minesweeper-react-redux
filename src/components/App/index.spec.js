import React from 'react'
import { shallow } from 'enzyme'

import App from '.'

describe('components:App', () => {
  it('renders the App with a blank game board and eighty-one squares', () => {
    expect(toJson(shallow(<App />).dive())).toMatchSnapshot()
  })

  it('renders the App with an isMine property passed down', () => {
    expect(
      toJson(shallow(<App isMine={true} />).dive())
    ).toMatchSnapshot()
  })

  // it('renders the App with a game board with three moves: squares 5, 18, and 49', () => {
  //   expect(toJson(shallow(<App moves={[5, 18, 49]} />).dive())).toMatchSnapshot()
  // })

  // it('renders the App with a handleClick passed down if the square is not in the moves array', () => {
  //   expect(toJson(shallow(<App handleClick={() => console.log(`Square ${square}`)} moves={[5, 18, 49]} />).dive())).toMatchSnapshot()
  // })
}) 