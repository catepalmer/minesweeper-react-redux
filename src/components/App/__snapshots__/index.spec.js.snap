// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:App renders the App with a blank game board and eighty-one squares 1`] = `
.c0 {
  display: grid;
  font-family: 'Verdana',sans-serif;
  grid-template-areas: 'board';
  height: 100vh;
  margin: 0;
  padding: 0;
  width: 100vw;
}

<div
  className="c0"
>
  <Board>
    <Square
      handleClick={[Function]}
      index={0}
      key="0"
    />
    <Square
      handleClick={[Function]}
      index={1}
      key="1"
    />
    <Square
      handleClick={[Function]}
      index={2}
      key="2"
    />
    <Square
      handleClick={[Function]}
      index={3}
      key="3"
    />
    <Square
      handleClick={[Function]}
      index={4}
      key="4"
    />
    <Square
      handleClick={[Function]}
      index={5}
      key="5"
    />
    <Square
      handleClick={[Function]}
      index={6}
      key="6"
    />
    <Square
      handleClick={[Function]}
      index={7}
      key="7"
    />
    <Square
      handleClick={[Function]}
      index={8}
      key="8"
    />
    <Square
      handleClick={[Function]}
      index={9}
      key="9"
    />
    <Square
      handleClick={[Function]}
      index={10}
      key="10"
    />
    <Square
      handleClick={[Function]}
      index={11}
      key="11"
    />
    <Square
      handleClick={[Function]}
      index={12}
      key="12"
    />
    <Square
      handleClick={[Function]}
      index={13}
      key="13"
    />
    <Square
      handleClick={[Function]}
      index={14}
      key="14"
    />
    <Square
      handleClick={[Function]}
      index={15}
      key="15"
    />
    <Square
      handleClick={[Function]}
      index={16}
      key="16"
    />
    <Square
      handleClick={[Function]}
      index={17}
      key="17"
    />
    <Square
      handleClick={[Function]}
      index={18}
      key="18"
    />
    <Square
      handleClick={[Function]}
      index={19}
      key="19"
    />
    <Square
      handleClick={[Function]}
      index={20}
      key="20"
    />
    <Square
      handleClick={[Function]}
      index={21}
      key="21"
    />
    <Square
      handleClick={[Function]}
      index={22}
      key="22"
    />
    <Square
      handleClick={[Function]}
      index={23}
      key="23"
    />
    <Square
      handleClick={[Function]}
      index={24}
      key="24"
    />
    <Square
      handleClick={[Function]}
      index={25}
      key="25"
    />
    <Square
      handleClick={[Function]}
      index={26}
      key="26"
    />
    <Square
      handleClick={[Function]}
      index={27}
      key="27"
    />
    <Square
      handleClick={[Function]}
      index={28}
      key="28"
    />
    <Square
      handleClick={[Function]}
      index={29}
      key="29"
    />
    <Square
      handleClick={[Function]}
      index={30}
      key="30"
    />
    <Square
      handleClick={[Function]}
      index={31}
      key="31"
    />
    <Square
      handleClick={[Function]}
      index={32}
      key="32"
    />
    <Square
      handleClick={[Function]}
      index={33}
      key="33"
    />
    <Square
      handleClick={[Function]}
      index={34}
      key="34"
    />
    <Square
      handleClick={[Function]}
      index={35}
      key="35"
    />
    <Square
      handleClick={[Function]}
      index={36}
      key="36"
    />
    <Square
      handleClick={[Function]}
      index={37}
      key="37"
    />
    <Square
      handleClick={[Function]}
      index={38}
      key="38"
    />
    <Square
      handleClick={[Function]}
      index={39}
      key="39"
    />
    <Square
      handleClick={[Function]}
      index={40}
      key="40"
    />
    <Square
      handleClick={[Function]}
      index={41}
      key="41"
    />
    <Square
      handleClick={[Function]}
      index={42}
      key="42"
    />
    <Square
      handleClick={[Function]}
      index={43}
      key="43"
    />
    <Square
      handleClick={[Function]}
      index={44}
      key="44"
    />
    <Square
      handleClick={[Function]}
      index={45}
      key="45"
    />
    <Square
      handleClick={[Function]}
      index={46}
      key="46"
    />
    <Square
      handleClick={[Function]}
      index={47}
      key="47"
    />
    <Square
      handleClick={[Function]}
      index={48}
      key="48"
    />
    <Square
      handleClick={[Function]}
      index={49}
      key="49"
    />
    <Square
      handleClick={[Function]}
      index={50}
      key="50"
    />
    <Square
      handleClick={[Function]}
      index={51}
      key="51"
    />
    <Square
      handleClick={[Function]}
      index={52}
      key="52"
    />
    <Square
      handleClick={[Function]}
      index={53}
      key="53"
    />
    <Square
      handleClick={[Function]}
      index={54}
      key="54"
    />
    <Square
      handleClick={[Function]}
      index={55}
      key="55"
    />
    <Square
      handleClick={[Function]}
      index={56}
      key="56"
    />
    <Square
      handleClick={[Function]}
      index={57}
      key="57"
    />
    <Square
      handleClick={[Function]}
      index={58}
      key="58"
    />
    <Square
      handleClick={[Function]}
      index={59}
      key="59"
    />
    <Square
      handleClick={[Function]}
      index={60}
      key="60"
    />
    <Square
      handleClick={[Function]}
      index={61}
      key="61"
    />
    <Square
      handleClick={[Function]}
      index={62}
      key="62"
    />
    <Square
      handleClick={[Function]}
      index={63}
      key="63"
    />
    <Square
      handleClick={[Function]}
      index={64}
      key="64"
    />
    <Square
      handleClick={[Function]}
      index={65}
      key="65"
    />
    <Square
      handleClick={[Function]}
      index={66}
      key="66"
    />
    <Square
      handleClick={[Function]}
      index={67}
      key="67"
    />
    <Square
      handleClick={[Function]}
      index={68}
      key="68"
    />
    <Square
      handleClick={[Function]}
      index={69}
      key="69"
    />
    <Square
      handleClick={[Function]}
      index={70}
      key="70"
    />
    <Square
      handleClick={[Function]}
      index={71}
      key="71"
    />
    <Square
      handleClick={[Function]}
      index={72}
      key="72"
    />
    <Square
      handleClick={[Function]}
      index={73}
      key="73"
    />
    <Square
      handleClick={[Function]}
      index={74}
      key="74"
    />
    <Square
      handleClick={[Function]}
      index={75}
      key="75"
    />
    <Square
      handleClick={[Function]}
      index={76}
      key="76"
    />
    <Square
      handleClick={[Function]}
      index={77}
      key="77"
    />
    <Square
      handleClick={[Function]}
      index={78}
      key="78"
    />
    <Square
      handleClick={[Function]}
      index={79}
      key="79"
    />
    <Square
      handleClick={[Function]}
      index={80}
      key="80"
    />
  </Board>
</div>
`;

exports[`components:App renders the App with an isMine property passed down 1`] = `
.c0 {
  display: grid;
  font-family: 'Verdana',sans-serif;
  grid-template-areas: 'board';
  height: 100vh;
  margin: 0;
  padding: 0;
  width: 100vw;
}

<div
  className="c0"
>
  <Board>
    <Square
      handleClick={[Function]}
      index={0}
      key="0"
    />
    <Square
      handleClick={[Function]}
      index={1}
      key="1"
    />
    <Square
      handleClick={[Function]}
      index={2}
      key="2"
    />
    <Square
      handleClick={[Function]}
      index={3}
      key="3"
    />
    <Square
      handleClick={[Function]}
      index={4}
      key="4"
    />
    <Square
      handleClick={[Function]}
      index={5}
      key="5"
    />
    <Square
      handleClick={[Function]}
      index={6}
      key="6"
    />
    <Square
      handleClick={[Function]}
      index={7}
      key="7"
    />
    <Square
      handleClick={[Function]}
      index={8}
      key="8"
    />
    <Square
      handleClick={[Function]}
      index={9}
      key="9"
    />
    <Square
      handleClick={[Function]}
      index={10}
      key="10"
    />
    <Square
      handleClick={[Function]}
      index={11}
      key="11"
    />
    <Square
      handleClick={[Function]}
      index={12}
      key="12"
    />
    <Square
      handleClick={[Function]}
      index={13}
      key="13"
    />
    <Square
      handleClick={[Function]}
      index={14}
      key="14"
    />
    <Square
      handleClick={[Function]}
      index={15}
      key="15"
    />
    <Square
      handleClick={[Function]}
      index={16}
      key="16"
    />
    <Square
      handleClick={[Function]}
      index={17}
      key="17"
    />
    <Square
      handleClick={[Function]}
      index={18}
      key="18"
    />
    <Square
      handleClick={[Function]}
      index={19}
      key="19"
    />
    <Square
      handleClick={[Function]}
      index={20}
      key="20"
    />
    <Square
      handleClick={[Function]}
      index={21}
      key="21"
    />
    <Square
      handleClick={[Function]}
      index={22}
      key="22"
    />
    <Square
      handleClick={[Function]}
      index={23}
      key="23"
    />
    <Square
      handleClick={[Function]}
      index={24}
      key="24"
    />
    <Square
      handleClick={[Function]}
      index={25}
      key="25"
    />
    <Square
      handleClick={[Function]}
      index={26}
      key="26"
    />
    <Square
      handleClick={[Function]}
      index={27}
      key="27"
    />
    <Square
      handleClick={[Function]}
      index={28}
      key="28"
    />
    <Square
      handleClick={[Function]}
      index={29}
      key="29"
    />
    <Square
      handleClick={[Function]}
      index={30}
      key="30"
    />
    <Square
      handleClick={[Function]}
      index={31}
      key="31"
    />
    <Square
      handleClick={[Function]}
      index={32}
      key="32"
    />
    <Square
      handleClick={[Function]}
      index={33}
      key="33"
    />
    <Square
      handleClick={[Function]}
      index={34}
      key="34"
    />
    <Square
      handleClick={[Function]}
      index={35}
      key="35"
    />
    <Square
      handleClick={[Function]}
      index={36}
      key="36"
    />
    <Square
      handleClick={[Function]}
      index={37}
      key="37"
    />
    <Square
      handleClick={[Function]}
      index={38}
      key="38"
    />
    <Square
      handleClick={[Function]}
      index={39}
      key="39"
    />
    <Square
      handleClick={[Function]}
      index={40}
      key="40"
    />
    <Square
      handleClick={[Function]}
      index={41}
      key="41"
    />
    <Square
      handleClick={[Function]}
      index={42}
      key="42"
    />
    <Square
      handleClick={[Function]}
      index={43}
      key="43"
    />
    <Square
      handleClick={[Function]}
      index={44}
      key="44"
    />
    <Square
      handleClick={[Function]}
      index={45}
      key="45"
    />
    <Square
      handleClick={[Function]}
      index={46}
      key="46"
    />
    <Square
      handleClick={[Function]}
      index={47}
      key="47"
    />
    <Square
      handleClick={[Function]}
      index={48}
      key="48"
    />
    <Square
      handleClick={[Function]}
      index={49}
      key="49"
    />
    <Square
      handleClick={[Function]}
      index={50}
      key="50"
    />
    <Square
      handleClick={[Function]}
      index={51}
      key="51"
    />
    <Square
      handleClick={[Function]}
      index={52}
      key="52"
    />
    <Square
      handleClick={[Function]}
      index={53}
      key="53"
    />
    <Square
      handleClick={[Function]}
      index={54}
      key="54"
    />
    <Square
      handleClick={[Function]}
      index={55}
      key="55"
    />
    <Square
      handleClick={[Function]}
      index={56}
      key="56"
    />
    <Square
      handleClick={[Function]}
      index={57}
      key="57"
    />
    <Square
      handleClick={[Function]}
      index={58}
      key="58"
    />
    <Square
      handleClick={[Function]}
      index={59}
      key="59"
    />
    <Square
      handleClick={[Function]}
      index={60}
      key="60"
    />
    <Square
      handleClick={[Function]}
      index={61}
      key="61"
    />
    <Square
      handleClick={[Function]}
      index={62}
      key="62"
    />
    <Square
      handleClick={[Function]}
      index={63}
      key="63"
    />
    <Square
      handleClick={[Function]}
      index={64}
      key="64"
    />
    <Square
      handleClick={[Function]}
      index={65}
      key="65"
    />
    <Square
      handleClick={[Function]}
      index={66}
      key="66"
    />
    <Square
      handleClick={[Function]}
      index={67}
      key="67"
    />
    <Square
      handleClick={[Function]}
      index={68}
      key="68"
    />
    <Square
      handleClick={[Function]}
      index={69}
      key="69"
    />
    <Square
      handleClick={[Function]}
      index={70}
      key="70"
    />
    <Square
      handleClick={[Function]}
      index={71}
      key="71"
    />
    <Square
      handleClick={[Function]}
      index={72}
      key="72"
    />
    <Square
      handleClick={[Function]}
      index={73}
      key="73"
    />
    <Square
      handleClick={[Function]}
      index={74}
      key="74"
    />
    <Square
      handleClick={[Function]}
      index={75}
      key="75"
    />
    <Square
      handleClick={[Function]}
      index={76}
      key="76"
    />
    <Square
      handleClick={[Function]}
      index={77}
      key="77"
    />
    <Square
      handleClick={[Function]}
      index={78}
      key="78"
    />
    <Square
      handleClick={[Function]}
      index={79}
      key="79"
    />
    <Square
      handleClick={[Function]}
      index={80}
      key="80"
    />
  </Board>
</div>
`;
